addons/point_of_sale/static/src/js/Popups/ClosePosPopup.js

```diff
@@ -0,0 +1,169 @@
+odoo.define('point_of_sale.ClosePosPopup', function(require) {
+    'use strict';
+
+    const { useState, useRef } = owl.hooks;
+    const AbstractAwaitablePopup = require('point_of_sale.AbstractAwaitablePopup');
+    const Registries = require('point_of_sale.Registries');
+    const { identifyError } = require('point_of_sale.utils');
+    const { ConnectionLostError, ConnectionAbortedError} = require('@web/core/network/rpc_service')
+    const { round_precision: round_pr } = require('web.utils');
+
+    /**
+     * This popup needs to be self-dependent because it needs to be called from different place.
+     */
+    class ClosePosPopup extends AbstractAwaitablePopup {
+        constructor() {
+            super(...arguments);
+            this.manualInputCashCount = true;
+            this.cashControl = this.env.pos.config.cash_control;
+            this.moneyDetailsRef = useRef('moneyDetails');
+            this.closeSessionClicked = false;
+            this.moneyDetails = null;
+        }
+        async willStart() {
+            try {
+                const closingData = await this.rpc({
+                    model: 'pos.session',
+                    method: 'get_closing_control_data',
+                    args: [[this.env.pos.pos_session.id]]
+                });
+                this.ordersDetails = closingData.orders_details;
+                this.paymentsAmount = closingData.payments_amount;
+                this.payLaterAmount = closingData.pay_later_amount;
+                this.openingNotes = closingData.opening_notes;
+                this.defaultCashDetails = closingData.default_cash_details;
+                this.otherPaymentMethods = closingData.other_payment_methods;
+
+                // component state and refs definition
+                const state = {notes: '', acceptClosing: false};
+                if (this.cashControl) {
+                    state[this.defaultCashDetails.name] = {counted: 0, difference: -this.defaultCashDetails.amount};
+                }
+                if (this.otherPaymentMethods.length > 0) {
+                    this.otherPaymentMethods.forEach(pm => {
+                        state[pm.name] = {counted: pm.amount, difference: 0}
```

Comment:
The input is always a number due to the `t-model.number`. The method `format_currency_no_symbol` will not work in some currencies such as dollars because it's a string that can contains coma `,`.

I have created a new method for rounding in the pos model which should do the trick:
```js
     round_decimals_currency(value) {
        const decimals = this.currency.decimals;
        return parseFloat(round_di(value, decimals).toFixed(decimals));
    },
```

Reply:
@trj-odoo don't use the input of type "number", it doesn't work properly in Odoo. Not even the float or int fields in backend are of this type.

Reply:
if you need to edit a number, the separator is not supposed to be the one of the currency, but the one set on the user. (or language, i don't remember). But I think it is already handled in the POS